packages/@adobe/spectrum-css-temp/components/dialog/index.css

```diff
@@ -134,6 +134,11 @@ governing permissions and limitations under the License.
   box-sizing: border-box;
 
   outline: none; /* Hide focus outline around header */
+
+  &.spectrum-Dialog-header--noTypeIcon {
+    grid-column-start: header;
+    grid-column-end: typeIcon;
```

Comment:
technically, we should apply this to heading as well
https://github.com/adobe/react-spectrum/blob/572a7cbf89ba2a2e3c6102685d57416c4aaf2ac5/packages/%40adobe/spectrum-css-temp/components/dialog/index.css#L121

If there is no header, then the heading takes over its grid area, which now might also include the typeIcon

Reply:
There are a lot of cases we aren't testing. I think the heading at line 341 works because in a lot of cases the forth column is collapsed to zero width meaning the typeIcon isn't in effect. To cause the 4th column to grow we probably need long footers or buttons to cause these.